client/app.jsx

```diff
@@ -12,31 +12,53 @@ export default class App extends React.Component {
     super(props);
     this.state = ({
       user: null,
+      newUser: false,
       isAuthorizing: true,
       route: parseRoute(window.location.hash)
     });
     this.handleSignIn = this.handleSignIn.bind(this);
     this.handleSignOut = this.handleSignOut.bind(this);
+    this.endTour = this.endTour.bind(this);
   }
 
   handleSignIn(result) {
     const { user, token } = result;
+    const { newUser } = jwtDecode(token);
     window.localStorage.setItem('user-token', token);
-    this.setState({ user, token });
+    window.localStorage.setItem('newUser', newUser);
+    this.setState({ user, token, newUser });
   }
 
   handleSignOut() {
     window.localStorage.removeItem('user-token');
     this.setState({ user: null });
   }
 
+  endTour() {
+    const { user } = this.state;
+    const req = {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json',
+        userId: user.userId
+      }
+    };
```

Comment:
🤔 We can't really allow the client to "tell us" things about the users after they've already signed up, unless they include a token. If your users have a token at this point, it should be included in this request and the server should require it.

Otherwise, I think the best way to approach this is to store a flag in localStorage, and treat that as the source of truth until the user actually signs in.

Reply:
Alright, totally missed this when I had added the userId on my requests but I stored the token from local storage into my state (then added to context). Then I send them in the request header for my authorization middleware to catch and verify the token.

![image](https://user-images.githubusercontent.com/109567499/202830057-a68ea226-9aad-4c69-8a14-70050dcd9979.png)

I placed all of my routes that require the authorization middleware below and then the auth ones that do not require a token are above the middleware.

![image](https://user-images.githubusercontent.com/109567499/202830067-efc747f5-6c9a-4860-9fa0-7351a20eeffd.png)